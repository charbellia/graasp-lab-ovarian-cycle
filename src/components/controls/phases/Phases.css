.Phases-Container {
  padding: 20px;
}
.select-step {
  font-size: 35px;
}

.active-ovulation, .active-preovulation {
  background-color: #6c757d;
  border-color: #6c757d;
  color: #fff;
}
