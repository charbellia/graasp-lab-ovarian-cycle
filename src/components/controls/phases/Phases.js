import React from 'react';
import './Phases.css';
import { Button } from 'reactstrap';

export const Phases = ({ seconds }) => (
  <div className="Phases-Container">
    <div className="phases-title">
      <span className="select-step"> Select step... </span>
    </div>
    <div className="phases">
      <Button outline color="secondary" className='pre-ovulation' >Pre-Ovulation</Button>
      <Button outline color="secondary" className='mx-2 ovulation' >Ovulation</Button>
      <Button outline color="secondary" className='post-ovulation' >Post-Ovulation</Button>
    </div>
  </div>
);

export default (Phases);
