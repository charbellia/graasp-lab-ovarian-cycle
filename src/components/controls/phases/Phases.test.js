import React from 'react';
import Phases from './Phases';

describe('render phase components tests', () => {
  it('has one instance of phase', () => {
    const props = {};
  });
});
