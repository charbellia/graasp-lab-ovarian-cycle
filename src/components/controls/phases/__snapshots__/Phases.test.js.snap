// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Phases /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Phases
    handleOvulation={[MockFunction]}
    handlePostOvulation={[MockFunction]}
    handlePreOvulation={[MockFunction]}
    ovulationActive={false}
    ovulationStep={false}
    postOvulationActive={false}
    postOvulationStep={false}
    preOvulation={false}
    preOvulationActive={false}
    preOvulationStep={false}
    t={
      [MockFunction] {
        "calls": Array [
          Array [
            "Pre-Ovulation",
          ],
          Array [
            "Ovulation",
          ],
          Array [
            "Post-Ovulation",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="phases"
      >
        <Button
          className="  cycle-phase"
          color="secondary"
          disabled={false}
          onClick={[MockFunction]}
          outline={true}
          tag="button"
        />
        <br />
        <Button
          className="  m-2 cycle-phase"
          color="secondary"
          disabled={false}
          onClick={[MockFunction]}
          outline={true}
          tag="button"
        />
        <br />
        <Button
          className="  cycle-phase"
          color="secondary"
          disabled={false}
          onClick={[MockFunction]}
          outline={true}
          tag="button"
        />
      </div>,
      "className": "phases-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Button
            className="  cycle-phase"
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            outline={true}
            tag="button"
          />,
          <br />,
          <Button
            className="  m-2 cycle-phase"
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            outline={true}
            tag="button"
          />,
          <br />,
          <Button
            className="  cycle-phase"
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            outline={true}
            tag="button"
          />,
        ],
        "className": "phases",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "className": "  cycle-phase",
            "color": "secondary",
            "disabled": false,
            "onClick": [MockFunction],
            "outline": true,
            "tag": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "className": "  m-2 cycle-phase",
            "color": "secondary",
            "disabled": false,
            "onClick": [MockFunction],
            "outline": true,
            "tag": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": undefined,
            "className": "  cycle-phase",
            "color": "secondary",
            "disabled": false,
            "onClick": [MockFunction],
            "outline": true,
            "tag": "button",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="phases"
        >
          <Button
            className="  cycle-phase"
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            outline={true}
            tag="button"
          />
          <br />
          <Button
            className="  m-2 cycle-phase"
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            outline={true}
            tag="button"
          />
          <br />
          <Button
            className="  cycle-phase"
            color="secondary"
            disabled={false}
            onClick={[MockFunction]}
            outline={true}
            tag="button"
          />
        </div>,
        "className": "phases-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              className="  cycle-phase"
              color="secondary"
              disabled={false}
              onClick={[MockFunction]}
              outline={true}
              tag="button"
            />,
            <br />,
            <Button
              className="  m-2 cycle-phase"
              color="secondary"
              disabled={false}
              onClick={[MockFunction]}
              outline={true}
              tag="button"
            />,
            <br />,
            <Button
              className="  cycle-phase"
              color="secondary"
              disabled={false}
              onClick={[MockFunction]}
              outline={true}
              tag="button"
            />,
          ],
          "className": "phases",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "className": "  cycle-phase",
              "color": "secondary",
              "disabled": false,
              "onClick": [MockFunction],
              "outline": true,
              "tag": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "className": "  m-2 cycle-phase",
              "color": "secondary",
              "disabled": false,
              "onClick": [MockFunction],
              "outline": true,
              "tag": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": undefined,
              "className": "  cycle-phase",
              "color": "secondary",
              "disabled": false,
              "onClick": [MockFunction],
              "outline": true,
              "tag": "button",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
